### PR TITLE
fi-a2-muhammad-bilal

### PR DESCRIPTION
**Pull request by devmbilal**
## Things you have improved in this PR

- UI has done.
- In this PR i do not break all the widgets due to time short but Inshallah will be done in next assesments.

## Checklist

- [x] Added PR title as per format i.e. `fi-a#-your-name`
- [x] Files and Folders names are per format i.e. `fi_a#_your_name_home.dart` and `FiA2YourNameHome`
- [ ] Get reviewer approval

## Screenshots (if applicable)
Screenshot_20230721-121035
Screenshot_20230721-112318 (1)
Screenshot_20230721-120441

### CODE REVIEW

File: assets/fi_a2_mbilal_assets/Add.png
Comment (mhmzdev) [inline assets/fi_a2_mbilal_assets/Add.png:1]: Please separate the svgs and pngs in your folder

File: pubspec.yaml [hunk 1]
```diff
@@ -64,6 +64,14 @@ flutter:
     - assets/
     - assets/svgs/
 
+
```
Comment (mhmzdev) [inline pubspec.yaml]: Remove this gap, its a lot of gap here

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 2]
```diff
@@ -0,0 +1,174 @@
...
+
+class BilalProfileScreen extends StatefulWidget {
+  const BilalProfileScreen({Key? key}) : super(key: key);
+
+  @override
+  State<BilalProfileScreen> createState() => _BilalProfileScreenState();
+}
+
+class _BilalProfileScreenState extends State<BilalProfileScreen> {
+  @override
```
Comment (mhmzdev) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: @override label is in the wrong place

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/fi_a2_mbilal_homescreen.dart [hunk 3]
```diff
@@ -0,0 +1,70 @@
...
+
+class BilalHomeScreen extends StatefulWidget {
+  const BilalHomeScreen({Key? key}) : super(key: key);
+
+  @override
+  State<BilalHomeScreen> createState() => _BilalHomeScreenState();
+}
+
+class _BilalHomeScreenState extends State<BilalHomeScreen> {
+  @override
```
Comment (mhmzdev) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/fi_a2_mbilal_homescreen.dart:16]: @override
label is in the wrong place

**Review by mhmzdev — CHANGES_REQUESTED** [review]
Pretty neat work. Good job, Minor fixes

File: pubspec.yaml (hunk 1 above)
Comment (devmbilal) [reply pubspec.yaml]: Resolved

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart (hunk 2 above)
Comment (devmbilal) [reply lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: Resolved

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/fi_a2_mbilal_homescreen.dart (hunk 3 above)
Comment (devmbilal) [reply lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/fi_a2_mbilal_homescreen.dart:16]: Resolved

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/fi_a2_mbilal_homescreen.dart [hunk 4]
```diff
@@ -0,0 +1,69 @@
+
+import 'package:flutter/material.dart';
+import 'package:flutter_interns/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_bottombar.dart';
+import 'package:flutter_interns/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart';
+import 'package:flutter_svg/flutter_svg.dart';
+
+class BilalHomeScreen extends StatefulWidget {
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/fi_a2_mbilal_homescreen.dart]: Why does this needs to stateful

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_bottombar.dart [hunk 5]
```diff
@@ -0,0 +1,34 @@
...
+class BilalBottomBar extends StatefulWidget {
+  const BilalBottomBar({Key? key}) : super(key: key);
+
+  @override
+  State<BilalBottomBar> createState() => _BilalBottomBarState();
+}
+
+class _BilalBottomBarState extends State<BilalBottomBar> {
+  @override
+  List<String> icons=['assets/svgs/home.svg','assets/svgs/Search.svg','assets/svgs/add.svg','assets/svgs/Profile.svg','assets/svgs/notification_2.svg'];
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_bottombar.dart]: please make this final

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart [hunk 6]
```diff
@@ -0,0 +1,125 @@
+
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+
+class BilalHomeContainer extends StatelessWidget {
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart:4]: Please name the class properly, HomeContainer does not describe the purpose of this

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_bottombar.dart [hunk 7]
```diff
@@ -0,0 +1,34 @@
+import 'package:flutter/material.dart';
+import 'package:flutter_svg/svg.dart';
+
+class BilalBottomBar extends StatefulWidget {
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_bottombar.dart]: Are you sure this needs to be stateful?

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart [hunk 8]
```diff
@@ -0,0 +1,125 @@
...
+                    border: Border.all(
+                      width: 3,
+                      color: Color(0xFFFA9884),
+                    ),
+
+                  ),
+                  child: CircleAvatar(
+                      backgroundColor: Colors.white,
+                      radius: 25,
+                      child: Image.asset('assets/fi_a2_mbilal_assets/man3.png',width: 45,height: 45,)),
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart]: Please add trailing commas and format the code

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart [hunk 9]
```diff
@@ -0,0 +1,125 @@
...
+                  SvgPicture.asset('assets/fi_a2_mbilal_assets/Like.svg',
+                    color: Colors.black,
+                    width: 20,height: 20,),
+                  const SizedBox(width: 7,),
+                  const Text('573',
+                    style: TextStyle(
+                      fontSize: 10,
+                      fontWeight: FontWeight.w600,
+                      fontFamily: 'Inter',
+                    ),),
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart]: format the code please

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 10]
```diff
@@ -0,0 +1,174 @@
...
+
+class BilalProfileScreen extends StatefulWidget {
+  const BilalProfileScreen({Key? key}) : super(key: key);
+
+  @override
+  State<BilalProfileScreen> createState() => _BilalProfileScreenState();
+}
+
+class _BilalProfileScreenState extends State<BilalProfileScreen> {
+  List<String> text1 =['Photos','Followers','Follows'];
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: please make this final

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 11]
```diff
@@ -0,0 +1,174 @@
...
+  List<String> text2 =['315','77.5k','500'];
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: please make this final

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 12]
```diff
@@ -0,0 +1,174 @@
...
+  List<String> text3=['Photos','Video','Tagged'];
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: please make this final

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 13]
```diff
@@ -0,0 +1,174 @@
...
+  List<String> picture=['assets/fi_a2_mbilal_assets/pic2.png','assets/fi_a2_mbilal_assets/pic3.png','assets/fi_a2_mbilal_assets/pic4.png'];
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: please make this final

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 14]
```diff
@@ -0,0 +1,174 @@
...
+             mainAxisAlignment: MainAxisAlignment.spaceBetween,
+             children: [
+               SvgPicture.asset('assets/fi_a2_mbilal_assets/Go back.svg',
+                 width: 27,height:19,),
+               const Text('My Profile',
+                 style: TextStyle(
+                   fontSize: 23,
+                   fontWeight: FontWeight.w700,
+                   fontFamily: 'Inter',
+                 ),),
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: please format the code.

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 15]
```diff
@@ -0,0 +1,174 @@
...
+               SvgPicture.asset('assets/fi_a2_mbilal_assets/Settings.svg',
+                 width: 27,height: 28,),
+             ],
+           ),
+         ),
+         const Padding(
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: Don't use padding for spaces

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart [hunk 16]
```diff
@@ -0,0 +1,174 @@
...
+           padding: EdgeInsets.only(top:40),
+           child: Center(
+             child: CircleAvatar(
+               backgroundImage:AssetImage('assets/girl1.png'),
+               radius: 50,
+             ),
+           ),
+         ),
+         const Padding(
+           padding: EdgeInsets.only(top:24),
```
Comment (ahmadexe) [inline lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: If this padding is for space pleas use a sizedBox

**Review by ahmadexe — CHANGES_REQUESTED** [review]
Please add trailing commas where required, and format

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart (hunk 8 above)
Comment (devmbilal) [reply lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_homescreen/widgets/fi_a2_mbilal_homecontainer.dart]: added

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart (hunk 10 above)
Comment (devmbilal) [reply lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: done

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart (hunk 13 above)
Comment (devmbilal) [reply lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: done

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart (hunk 12 above)
Comment (devmbilal) [reply lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: done

File: lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart (hunk 16 above)
Comment (devmbilal) [reply lib/assesments/assessment_2/fi_a2_muhammad_bilal/fi_a2_mbilal_profile/fi_a2_mbilal_profile.dart]: ok

**Pull request merged**